pkg/collector/corechecks/systemd/systemd.go

```diff
@@ -432,14 +457,18 @@ func getPropertyBool(properties map[string]interface{}, propertyName string) (bo
 	return propValue, nil
 }
 
-func getServiceCheckStatus(activeState string) metrics.ServiceCheckStatus {
-	switch activeState {
-	case "active":
+func getServiceCheckStatus(state string, substateMapping map[string]string) metrics.ServiceCheckStatus {
+	mappedServiceCheckStatus, ok := substateMapping[state]
+	if !ok {
+		return metrics.ServiceCheckUnknown
+	}
+	switch mappedServiceCheckStatus {
+	case "ok":
 		return metrics.ServiceCheckOK
-	case "inactive", "failed":
+	case "warning":
+		return metrics.ServiceCheckWarning
```

Comment:
This will never occur, no?

Reply:
Might occur for user provided mapping, example:

```
substate_status_mapping:
  foo:
    running: ok
    exited: warning
    mounted: unknown
```